File: ready-set-boole_rust/src/bin/exercices/ex00.rs
pub fn adder(a: u32, b: u32) -> u32 {
    let mut a = a;
    let mut b = b;
    let mut tmp;
    while b != 0 {
        tmp = a ^ b;
        b = (a & b) << 1;
        a = tmp;
    }
    return a;
}

File: ready-set-boole_java/fr/rvander/ready_set_boole/Ex00.java
package fr.rvander.ready_set_boole;


public class Ex00 {

	public static int adder(int a, int b) {
		int tmp;
		while (b != 0) {
			tmp = a ^ b;
			b = (a & b) << 1;
			a = tmp;
		}
		return a;
	}
}